### DISCUSSION TITLE
A cycle was detected when generating the classpath

### DISCUSSION THREAD

**Discussion by zhangxian314** [dbeaver/dbeaver#19895] [DBeaver Development]
### Description

when i export an eclipse product. 
 'Export Product' has encountered a problem.
A cycle was detected when generating the classpath org.jkiss.dbeaver.ext.greenplum_1.0.105.202305061143, org.jkiss.dbeaver.ext.postgresql_2.1.177.202305061143, org.jkiss.dbeaver.ext.greenplum_1.0.105.202305061143.

### DBeaver Version

23.0.2

### Operating System

windows 10 64bit

**Comment by chirontt** [comment]
Yes, I've encountered this cyclic-dependency problem in Eclipse when trying to do 'Export Product' of the DBeaver project. I believe this 'Export Product' is the old way of producing DBeaver (or any Eclipse-based application) binaries, although it should have worked.

Nowadays you should try producing the DBeaver application using the Maven command: `mvn package` (for `x86_64` binaries), or `mvn package -Pall-platforms` (for all binaries including those for the `aarch64` architecture)
